src/backend/distributed/operations/shard_cleaner.c

```diff
@@ -756,10 +756,16 @@ TryDropShardOutsideTransaction(OperationId operationId, char *qualifiedTableName
 									   dropQuery->data);
 
 	/* remove the shard from the node */
-	bool success = SendOptionalCommandListToWorkerOutsideTransaction(nodeName,
-																	 nodePort,
-																	 NULL,
-																	 dropCommandList);
+	int connectionFlags = OUTSIDE_TRANSACTION;
+	MultiConnection *workerConnection = GetNodeUserDatabaseConnection(connectionFlags,
+																	  nodeName, nodePort,
+																	  CurrentUserName(),
+																	  NULL);
+	workerConnection->forceCloseAtTransactionEnd = true;
```

Comment:
these might be useful connections to worker nodes

now I kind of wonder whether we should handle this a bit differently and explicitly check for localhost in ShouldShutdownConnection

Reply:
@onderkalaci you removed this in your WIP commit. Was that on purpose?

Reply:
Yes, it was on purpose. We cannot have that, see: https://github.com/citusdata/citus/pull/6314#discussion_r971929086

And, I think that is a must have, otherwise it is trivial to get connection errors.

I see two options:
- Explicitly close the connection with CloseConnection once we are all done (preferred)
- Let the connection stay (as Marco noted, I also think this can be ok for now given that we should only cache 1 connection to localhost)